lib/galaxy/model/__init__.py

```diff
@@ -1614,9 +1772,18 @@ def add_datasets(self, sa_session, datasets, parent_id=None, genome_build=None,
         optimize = len(datasets) > 1 and parent_id is None and all_hdas and set_hid
         if optimize:
             self.__add_datasets_optimized(datasets, genome_build=genome_build)
-            if quota and self.user:
-                disk_usage = sum([d.get_total_size() for d in datasets])
-                self.user.adjust_total_disk_usage(disk_usage)
+            if self.user:
```

Comment:
This piece of code was highly optimized... I would be sad if this were merged without verifying there isn't a huge performance hit here.

Reply:
How do you evaluate your performance gain? and how much reduction is acceptable?

Reply:
I guess this point can be considered resolved as per discussion at https://github.com/galaxyproject/galaxy/issues/8790